lib/logstash/inputs/jms.rb

```diff
@@ -122,6 +133,15 @@ class LogStash::Inputs::Jms < LogStash::Inputs::Threadable
   config :truststore, :validate => :path
   config :truststore_password, :validate => :password
 
+  # Defines a target field for placing fields.
+  # If this setting is omitted, data gets stored at the root (top level) of the event.
+  # The target is only relevant while decoding data into a new event.
+  #
+  # NOTE: this is only relevant for map messages, byte[] and string use the codec!
+  config :target, :validate => :field_reference
```

Comment:
considered using a `map_target` but we follow the `target` naming convention to be able to use `targeted_event_factory`